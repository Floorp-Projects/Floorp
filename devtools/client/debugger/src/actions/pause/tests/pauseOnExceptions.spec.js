/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

import {
  actions,
  createStore,
  getTelemetryEvents,
} from "../../../utils/test-head";

import {
  getShouldPauseOnExceptions,
  getShouldPauseOnCaughtExceptions,
} from "../../../selectors/pause";

describe("pauseOnExceptions", () => {
  it("should track telemetry for pauseOnException changes", async () => {
    const { dispatch, getState } = createStore({ pauseOnExceptions: () => {} });
    dispatch(actions.pauseOnExceptions(true, false));
    expect(getTelemetryEvents("pause_on_exceptions")).toMatchSnapshot();
    expect(getShouldPauseOnExceptions(getState())).toBe(true);
    expect(getShouldPauseOnCaughtExceptions(getState())).toBe(false);
  });
});
