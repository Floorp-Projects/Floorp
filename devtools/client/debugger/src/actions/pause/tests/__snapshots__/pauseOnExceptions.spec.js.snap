// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pauseOnExceptions should track telemetry for pauseOnException changes 1`] = `
Array [
  Object {
    "caught_exceptio": false,
    "exceptions": true,
  },
]
`;
